### PR TITLE
Remove fieldType() from MapperService; move IndexAnalyzer to IndexShard

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: server/src/main/java/io/crate/lucene/LuceneQueryBuilder.java [hunk 1]
```diff
@@ -203,13 +203,8 @@ public SymbolVisitor<Context, Query> visitor() {
...
         @Nullable
-        public Reference getRef(String field) {
-            try {
-                long oid = Long.parseLong(field);
-                return table.getReference(oid);
-            } catch (NumberFormatException ex) {
-                return getRef(ColumnIdent.fromPath(field));
-            }
+        public Reference getRef(String storageIdent) {
+            return table.getReference(storageIdent);
```
Comment (BaurzhanSakhariev) [inline server/src/main/java/io/crate/lucene/LuceneQueryBuilder.java:207]: I wonder if can rely on version.created instead of catching an exception.

Comment (BaurzhanSakhariev) [reply server/src/main/java/io/crate/lucene/LuceneQueryBuilder.java:207]: OK, looks like version wouldn't helps for all cases but we already have `lookupNameBySourceKey`?

**Review by BaurzhanSakhariev — APPROVED** [review]
Added a suggestion, LGTM otherwise

File: server/src/main/java/io/crate/lucene/LuceneQueryBuilder.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/lucene/LuceneQueryBuilder.java:207]: Yes I'm also not happy about the `O(n)` lookup but I'd like to keep it for now and follow up on it later to unblock the other work that builds on the changes here.

**Review by romseygeek — APPROVED** [review]
LGTM

**Pull request merged**